<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 QIIME 2 | HPC Tutorial in CINEMA</title>
  <meta name="description" content="This document is a concise guide for CINEMA Research Group members on using the HPC resources. It includes instructions for accessing the HPC cluster, submitting SLURM batch jobs, and using key bioinformatics tools like Woltka, Samtools, Qiime 2, and Kraken2, ensuring efficient utilization of HPC for research tasks." />
  <meta name="generator" content="bookdown 0.40 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 QIIME 2 | HPC Tutorial in CINEMA" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This document is a concise guide for CINEMA Research Group members on using the HPC resources. It includes instructions for accessing the HPC cluster, submitting SLURM batch jobs, and using key bioinformatics tools like Woltka, Samtools, Qiime 2, and Kraken2, ensuring efficient utilization of HPC for research tasks." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 QIIME 2 | HPC Tutorial in CINEMA" />
  
  <meta name="twitter:description" content="This document is a concise guide for CINEMA Research Group members on using the HPC resources. It includes instructions for accessing the HPC cluster, submitting SLURM batch jobs, and using key bioinformatics tools like Woltka, Samtools, Qiime 2, and Kraken2, ensuring efficient utilization of HPC for research tasks." />
  

<meta name="author" content="Yiqian Zhang" />


<meta name="date" content="2024-07-24" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="samtools.html"/>
<link rel="next" href="kraken2.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">HPC Tutorial in CINEMA</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>CINEMA Research Group</a></li>
<li class="chapter" data-level="1" data-path="getting-started.html"><a href="getting-started.html"><i class="fa fa-check"></i><b>1</b> Getting Started</a>
<ul>
<li class="chapter" data-level="1.1" data-path="getting-started.html"><a href="getting-started.html#lab-group-directory"><i class="fa fa-check"></i><b>1.1</b> Lab Group Directory</a></li>
<li class="chapter" data-level="1.2" data-path="getting-started.html"><a href="getting-started.html#key-areas-in-cwru-hpc-ondemand-web-portal"><i class="fa fa-check"></i><b>1.2</b> Key Areas in CWRU HPC OnDemand Web Portal</a></li>
<li class="chapter" data-level="1.3" data-path="getting-started.html"><a href="getting-started.html#bioinformatics-package-installation-status"><i class="fa fa-check"></i><b>1.3</b> Bioinformatics Package Installation Status</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="submitting-batch-jobs.html"><a href="submitting-batch-jobs.html"><i class="fa fa-check"></i><b>2</b> Submitting Batch Jobs</a></li>
<li class="chapter" data-level="3" data-path="conda.html"><a href="conda.html"><i class="fa fa-check"></i><b>3</b> Conda</a></li>
<li class="chapter" data-level="4" data-path="woltka.html"><a href="woltka.html"><i class="fa fa-check"></i><b>4</b> Woltka</a></li>
<li class="chapter" data-level="5" data-path="samtools.html"><a href="samtools.html"><i class="fa fa-check"></i><b>5</b> Samtools</a>
<ul>
<li class="chapter" data-level="5.1" data-path="samtools.html"><a href="samtools.html#use-samtools"><i class="fa fa-check"></i><b>5.1</b> Use Samtools</a></li>
<li class="chapter" data-level="5.2" data-path="samtools.html"><a href="samtools.html#install-samtools"><i class="fa fa-check"></i><b>5.2</b> Install Samtools</a></li>
<li class="chapter" data-level="5.3" data-path="samtools.html"><a href="samtools.html#convert-bam-to-fastq"><i class="fa fa-check"></i><b>5.3</b> Convert BAM to FASTQ</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="qiime-2.html"><a href="qiime-2.html"><i class="fa fa-check"></i><b>6</b> QIIME 2</a>
<ul>
<li class="chapter" data-level="6.1" data-path="qiime-2.html"><a href="qiime-2.html#use-qiime-2"><i class="fa fa-check"></i><b>6.1</b> Use QIIME 2</a></li>
<li class="chapter" data-level="6.2" data-path="qiime-2.html"><a href="qiime-2.html#available-qiime-2-databases"><i class="fa fa-check"></i><b>6.2</b> Available Qiime 2 Databases</a></li>
<li class="chapter" data-level="6.3" data-path="qiime-2.html"><a href="qiime-2.html#example-for-jax.iu.pitt-microbiome-pilot-study"><i class="fa fa-check"></i><b>6.3</b> Example for Jax.IU.Pitt Microbiome Pilot Study</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="qiime-2.html"><a href="qiime-2.html#importing-data"><i class="fa fa-check"></i><b>6.3.1</b> Importing Data</a></li>
<li class="chapter" data-level="6.3.2" data-path="qiime-2.html"><a href="qiime-2.html#trimming-adapters-from-sequences"><i class="fa fa-check"></i><b>6.3.2</b> Trimming Adapters from Sequences</a></li>
<li class="chapter" data-level="6.3.3" data-path="qiime-2.html"><a href="qiime-2.html#summarizing-demultiplexed-sequences"><i class="fa fa-check"></i><b>6.3.3</b> Summarizing Demultiplexed Sequences</a></li>
<li class="chapter" data-level="6.3.4" data-path="qiime-2.html"><a href="qiime-2.html#denoising-sequences-with-dada2"><i class="fa fa-check"></i><b>6.3.4</b> Denoising Sequences with DADA2</a></li>
<li class="chapter" data-level="6.3.5" data-path="qiime-2.html"><a href="qiime-2.html#taxonomic-classification"><i class="fa fa-check"></i><b>6.3.5</b> Taxonomic Classification</a></li>
<li class="chapter" data-level="6.3.6" data-path="qiime-2.html"><a href="qiime-2.html#metadata-visualization"><i class="fa fa-check"></i><b>6.3.6</b> Metadata Visualization</a></li>
<li class="chapter" data-level="6.3.7" data-path="qiime-2.html"><a href="qiime-2.html#feature-table-summary"><i class="fa fa-check"></i><b>6.3.7</b> Feature Table Summary</a></li>
<li class="chapter" data-level="6.3.8" data-path="qiime-2.html"><a href="qiime-2.html#phylogenetic-tree-construction"><i class="fa fa-check"></i><b>6.3.8</b> Phylogenetic Tree Construction</a></li>
<li class="chapter" data-level="6.3.9" data-path="qiime-2.html"><a href="qiime-2.html#taxa-bar-plot-visualization"><i class="fa fa-check"></i><b>6.3.9</b> Taxa Bar Plot Visualization</a></li>
<li class="chapter" data-level="6.3.10" data-path="qiime-2.html"><a href="qiime-2.html#alpha-rarefaction-analysis"><i class="fa fa-check"></i><b>6.3.10</b> Alpha Rarefaction Analysis</a></li>
<li class="chapter" data-level="6.3.11" data-path="qiime-2.html"><a href="qiime-2.html#core-metrics-phylogenetic-analysis"><i class="fa fa-check"></i><b>6.3.11</b> Core Metrics Phylogenetic Analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="kraken2.html"><a href="kraken2.html"><i class="fa fa-check"></i><b>7</b> Kraken2</a>
<ul>
<li class="chapter" data-level="7.1" data-path="kraken2.html"><a href="kraken2.html#use-kraken2"><i class="fa fa-check"></i><b>7.1</b> Use Kraken2</a></li>
<li class="chapter" data-level="7.2" data-path="kraken2.html"><a href="kraken2.html#available-kraken2-databases"><i class="fa fa-check"></i><b>7.2</b> Available Kraken2 Databases</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">HPC Tutorial in CINEMA</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="qiime-2" class="section level1 hasAnchor" number="6">
<h1><span class="header-section-number">Chapter 6</span> QIIME 2<a href="qiime-2.html#qiime-2" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><a href="https://qiime2.org/">QIIME 2 Website</a></p>
<p><strong>Qiime 2</strong> (Quantitative Insights Into Microbial Ecology 2) is a powerful, open-source bioinformatics platform designed for performing microbiome analysis from raw DNA sequencing data. It provides tools for the analysis and interpretation of high-throughput community sequencing data, facilitating tasks such as quality control, taxonomic classification, diversity analysis, and visualization. Qiime 2 supports a modular plugin system, enabling users to integrate various analytical methods and customize workflows, making it an essential tool for researchers studying microbial ecology, diversity, and function.</p>
<div id="use-qiime-2" class="section level2 hasAnchor" number="6.1">
<h2><span class="header-section-number">6.1</span> Use QIIME 2<a href="qiime-2.html#use-qiime-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Qiime 2 is already installed in the Lab Group Conda environment. To use it, follow these steps:</p>
<ol style="list-style-type: decimal">
<li>Activate Conda:
Run the following command to activate Conda:</li>
</ol>
<div class="sourceCode" id="cb16"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb16-1"><a href="qiime-2.html#cb16-1" tabindex="-1"></a><span class="bu">source</span> /mnt/vstor/SOM_PQHS_LXZ716/software/miniconda3/bin/activate</span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Activate the Qiime 2 Environment:
Activate the Qiime 2 environment by running:</li>
</ol>
<div class="sourceCode" id="cb17"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb17-1"><a href="qiime-2.html#cb17-1" tabindex="-1"></a><span class="ex">conda</span> activate qiime2-amplicon-2024.5</span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>Check the Version:
Run the following command to check if Qiime 2 is installed and to see the version number:</li>
</ol>
<div class="sourceCode" id="cb18"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb18-1"><a href="qiime-2.html#cb18-1" tabindex="-1"></a><span class="ex">qiime</span> <span class="at">--version</span></span></code></pre></div>
</div>
<div id="available-qiime-2-databases" class="section level2 hasAnchor" number="6.2">
<h2><span class="header-section-number">6.2</span> Available Qiime 2 Databases<a href="qiime-2.html#available-qiime-2-databases" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>A pre-trained classifier database for Qiime 2 is available in the Group Directory. This database can be found under the following directory:</p>
<p><code>/mnt/vstor/SOM_PQHS_LXZ716/Reference</code></p>
<p>The available database includes:</p>
<ul>
<li><code>QIIME2_Pre-trained_Classifier</code>: A pre-trained classifier for taxonomic classification of microbial sequences, optimized for use with Qiime 2.</li>
</ul>
</div>
<div id="example-for-jax.iu.pitt-microbiome-pilot-study" class="section level2 hasAnchor" number="6.3">
<h2><span class="header-section-number">6.3</span> Example for <a href="https://ondemand-pioneer.case.edu/pun/sys/dashboard/files/fs//mnt/vstor/SOM_PQHS_LXZ716/MicroB2">Jax.IU.Pitt Microbiome Pilot Study</a><a href="qiime-2.html#example-for-jax.iu.pitt-microbiome-pilot-study" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="importing-data" class="section level3 hasAnchor" number="6.3.1">
<h3><span class="header-section-number">6.3.1</span> Importing Data<a href="qiime-2.html#importing-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Fastq Manifest Formats</strong></p>
<p>When importing data, we use the “Fastq manifest” format. This method is suitable for data that do not fit the common multiplexed or demultiplexed formats (e.g., EMP or Casava). To import your data manually, you will need to create a manifest file and use the <code>qiime tools import</code> command.</p>
<p><strong>Format Description</strong></p>
<p>First, you’ll create a text file called a “manifest file”, which maps sample identifiers to fastq.gz or fastq absolute filepaths that contain sequence and quality data for the sample (i.e., these are FASTQ files). The manifest file also indicates the direction of the reads in each fastq.gz or fastq file. The manifest file is designed to be a simple format that doesn’t put restrictions on the naming of the demultiplexed fastq.gz / fastq files.</p>
<p>The manifest file is a tab-separated (i.e., .tsv) text file. The first column defines the Sample ID, while the second (and optional third) column defines the absolute filepath to the forward (and optional reverse) reads. The fastq.gz absolute filepaths may contain environment variables (e.g., $HOME or $PWD).</p>
<p><strong>Example Manifest File for Paired-End Reads</strong></p>
<p>Here is an example of a manifest file for paired-end read data:</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb19-1"><a href="qiime-2.html#cb19-1" tabindex="-1"></a><span class="ex">sample-id</span>    forward-absolute-filepath                    reverse-absolute-filepath</span>
<span id="cb19-2"><a href="qiime-2.html#cb19-2" tabindex="-1"></a><span class="ex">288593663</span>    /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/288593663_R1.fastq.gz    /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/288593663_R2.fastq.gz</span>
<span id="cb19-3"><a href="qiime-2.html#cb19-3" tabindex="-1"></a><span class="ex">288601191</span>    /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/288601191_R1.fastq.gz    /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/288601191_R2.fastq.gz</span>
<span id="cb19-4"><a href="qiime-2.html#cb19-4" tabindex="-1"></a><span class="ex">288605432</span>    /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/288605432_R1.fastq.gz    /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/288605432_R2.fastq.gz</span>
<span id="cb19-5"><a href="qiime-2.html#cb19-5" tabindex="-1"></a><span class="ex">288613647</span>    /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/288613647_R1.fastq.gz    /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/288613647_R2.fastq.gz</span>
<span id="cb19-6"><a href="qiime-2.html#cb19-6" tabindex="-1"></a><span class="ex">288625970</span>    /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/288625970_R1.fastq.gz    /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/288625970_R2.fastq.gz</span>
<span id="cb19-7"><a href="qiime-2.html#cb19-7" tabindex="-1"></a><span class="ex">288638583</span>    /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/288638583_R1.fastq.gz    /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/288638583_R2.fastq.gz</span>
<span id="cb19-8"><a href="qiime-2.html#cb19-8" tabindex="-1"></a><span class="ex">288638603</span>    /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/288638603_R1.fastq.gz    /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/288638603_R2.fastq.gz</span>
<span id="cb19-9"><a href="qiime-2.html#cb19-9" tabindex="-1"></a><span class="ex">288658590</span>    /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/288658590_R1.fastq.gz    /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/288658590_R2.fastq.gz</span>
<span id="cb19-10"><a href="qiime-2.html#cb19-10" tabindex="-1"></a><span class="ex">288662694</span>    /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/288662694_R1.fastq.gz    /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/288662694_R2.fastq.gz</span>
<span id="cb19-11"><a href="qiime-2.html#cb19-11" tabindex="-1"></a><span class="ex">288696281</span>    /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/288696281_R1.fastq.gz    /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/288696281_R2.fastq.gz</span></code></pre></div>
<p><strong>Importing the Data</strong></p>
<p>Use the following command to import the data into QIIME 2:</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb20-1"><a href="qiime-2.html#cb20-1" tabindex="-1"></a><span class="ex">qiime</span> tools import <span class="dt">\</span></span>
<span id="cb20-2"><a href="qiime-2.html#cb20-2" tabindex="-1"></a>  <span class="at">--type</span> <span class="st">&#39;SampleData[PairedEndSequencesWithQuality]&#39;</span> <span class="dt">\</span></span>
<span id="cb20-3"><a href="qiime-2.html#cb20-3" tabindex="-1"></a>  <span class="at">--input-path</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/manifest.tsv <span class="dt">\</span></span>
<span id="cb20-4"><a href="qiime-2.html#cb20-4" tabindex="-1"></a>  <span class="at">--output-path</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/paired-end-demux.qza <span class="dt">\</span></span>
<span id="cb20-5"><a href="qiime-2.html#cb20-5" tabindex="-1"></a>  <span class="at">--input-format</span> PairedEndFastqManifestPhred33V2</span></code></pre></div>
<p>For more information on importing data and using different formats, refer to the <a href="https://docs.qiime2.org/2024.5/tutorials/importing/">QIIME 2 Importing Tutorial</a>.</p>
</div>
<div id="trimming-adapters-from-sequences" class="section level3 hasAnchor" number="6.3.2">
<h3><span class="header-section-number">6.3.2</span> Trimming Adapters from Sequences<a href="qiime-2.html#trimming-adapters-from-sequences" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>After importing the data, it is important to remove any adapter sequences that may be present. The trimmed sequences can often be found in the STUDY DESCRIPTION. For example, in our study, the STUDY DESCRIPTION mentions:</p>
<p><code>The primers used incorporated Illumina dual indices and sequencing adapters with the following 16S rRNA gene V1-V3 priming regions: 27F (5’ AGAGTTTGATCCTGGCTCAG) and 534R (5’ ATTACCGCGGCTGCTGG).</code></p>
<p>Use the following command to trim adapters from the paired-end sequences:</p>
<p><a href="https://docs.qiime2.org/2024.5/plugins/available/cutadapt/trim-paired/">Qiime 2 Dosc</a></p>
<div class="sourceCode" id="cb21"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb21-1"><a href="qiime-2.html#cb21-1" tabindex="-1"></a><span class="ex">qiime</span> cutadapt trim-paired <span class="dt">\</span></span>
<span id="cb21-2"><a href="qiime-2.html#cb21-2" tabindex="-1"></a>  <span class="at">--i-demultiplexed-sequences</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/paired-end-demux.qza <span class="dt">\</span></span>
<span id="cb21-3"><a href="qiime-2.html#cb21-3" tabindex="-1"></a>  <span class="at">--p-cores</span> 16 <span class="dt">\</span></span>
<span id="cb21-4"><a href="qiime-2.html#cb21-4" tabindex="-1"></a>  <span class="at">--p-front-f</span> AGAGTTTGATCCTGGCTCAG <span class="dt">\</span></span>
<span id="cb21-5"><a href="qiime-2.html#cb21-5" tabindex="-1"></a>  <span class="at">--p-front-r</span> ATTACCGCGGCTGCTGG <span class="dt">\</span></span>
<span id="cb21-6"><a href="qiime-2.html#cb21-6" tabindex="-1"></a>  <span class="at">--p-discard-untrimmed</span> <span class="dt">\</span></span>
<span id="cb21-7"><a href="qiime-2.html#cb21-7" tabindex="-1"></a>  <span class="at">--o-trimmed-sequences</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/trim-seqs.qza <span class="dt">\</span></span>
<span id="cb21-8"><a href="qiime-2.html#cb21-8" tabindex="-1"></a>  <span class="at">--verbose</span></span></code></pre></div>
<ul>
<li><code>--i-demultiplexed-sequences</code>: Input file with demultiplexed sequences.</li>
<li><code>--p-cores</code>: Number of CPU cores to use for processing.</li>
<li><code>--p-front-f</code>: Forward adapter sequence to trim.</li>
<li><code>--p-front-r</code>: Reverse adapter sequence to trim.</li>
<li><code>--p-discard-untrimmed</code>: Discard reads that do not contain the adapter sequences.</li>
<li><code>--o-trimmed-sequences</code>: Output file for the trimmed sequences.</li>
<li><code>--verbose</code>: Provides detailed output during the trimming process.</li>
</ul>
<p>After trimming the adapters, proceed to the summarizing demultiplexed sequences step to visualize the quality of the trimmed data.</p>
</div>
<div id="summarizing-demultiplexed-sequences" class="section level3 hasAnchor" number="6.3.3">
<h3><span class="header-section-number">6.3.3</span> Summarizing Demultiplexed Sequences<a href="qiime-2.html#summarizing-demultiplexed-sequences" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Before denoising, it is important to visualize the quality of the sequencing data. This step summarizes the demultiplexed sequences and generates a visualization to help determine the optimal truncation lengths for denoising.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb22-1"><a href="qiime-2.html#cb22-1" tabindex="-1"></a><span class="ex">qiime</span> demux summarize <span class="dt">\</span></span>
<span id="cb22-2"><a href="qiime-2.html#cb22-2" tabindex="-1"></a>  <span class="at">--i-data</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/trim-seqs.qza <span class="dt">\</span></span>
<span id="cb22-3"><a href="qiime-2.html#cb22-3" tabindex="-1"></a>  <span class="at">--o-visualization</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/trim-seqs.qzv</span></code></pre></div>
<ul>
<li><code>--i-data</code>: Input file with demultiplexed sequences.</li>
<li><code>--o-visualization</code>: Output file for the summary visualization.</li>
</ul>
<p>The output visualization (<code>demux-summary.qzv</code>) provides interactive plots of the quality scores across all sequences. This information is crucial for determining the appropriate truncation lengths (<code>--p-trunc-len-f</code> and <code>--p-trunc-len-r</code>) to use in the denoising step.</p>
<div class="float">
<img src="demux_summary.png" alt="demux-summary.qzv" />
<div class="figcaption">demux-summary.qzv</div>
</div>
<p><strong>How to Choose Truncation Lengths:</strong></p>
<ol style="list-style-type: decimal">
<li>Open the Visualization:</li>
</ol>
<p>Open the demux-summary.qzv file in QIIME 2 View QIIME 2 View.</p>
<ol start="2" style="list-style-type: decimal">
<li>Inspect the Quality Plots:</li>
</ol>
<p>Look at the interactive quality plots for the forward and reverse reads. Each plot shows the distribution of quality scores across all positions in your reads.</p>
<ol start="3" style="list-style-type: decimal">
<li>Identify Quality Drop-off Points:</li>
</ol>
<p>Identify the position in the reads where the quality scores start to drop significantly. This is usually seen as a downward trend in the quality scores towards the end of the reads.
For example, if the quality scores drop below a threshold (e.g., Q25) around position 270 for the forward reads and position 260 for the reverse reads, these positions might be suitable truncation points.</p>
<ol start="3" style="list-style-type: decimal">
<li>Choose Conservative Truncation Lengths:</li>
</ol>
<p>To ensure high-quality data, choose truncation lengths that are slightly before the quality scores drop significantly. This helps to retain more high-quality bases.
In this example, you might choose –p-trunc-len-f 270 for the forward reads and –p-trunc-len-r 260 for the reverse reads.</p>
</div>
<div id="denoising-sequences-with-dada2" class="section level3 hasAnchor" number="6.3.4">
<h3><span class="header-section-number">6.3.4</span> Denoising Sequences with DADA2<a href="qiime-2.html#denoising-sequences-with-dada2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This step removes noise from the sequencing data, corrects errors, and generates a table of feature abundances (ASVs), representative sequences, and denoising statistics.</p>
<p><a href="https://docs.qiime2.org/2024.5/plugins/available/dada2/denoise-paired/">Qiime 2 Docs</a></p>
<div class="sourceCode" id="cb23"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb23-1"><a href="qiime-2.html#cb23-1" tabindex="-1"></a><span class="ex">qiime</span> dada2 denoise-paired <span class="dt">\</span></span>
<span id="cb23-2"><a href="qiime-2.html#cb23-2" tabindex="-1"></a>  <span class="at">--i-demultiplexed-seqs</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/trim-seqs.qza <span class="dt">\</span></span>
<span id="cb23-3"><a href="qiime-2.html#cb23-3" tabindex="-1"></a>  <span class="at">--p-trim-left-f</span> 0 <span class="dt">\</span></span>
<span id="cb23-4"><a href="qiime-2.html#cb23-4" tabindex="-1"></a>  <span class="at">--p-trim-left-r</span> 0 <span class="dt">\</span></span>
<span id="cb23-5"><a href="qiime-2.html#cb23-5" tabindex="-1"></a>  <span class="at">--p-trunc-len-f</span> 270 <span class="dt">\</span></span>
<span id="cb23-6"><a href="qiime-2.html#cb23-6" tabindex="-1"></a>  <span class="at">--p-trunc-len-r</span> 260 <span class="dt">\</span></span>
<span id="cb23-7"><a href="qiime-2.html#cb23-7" tabindex="-1"></a>  <span class="at">--o-table</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/table.qza <span class="dt">\</span></span>
<span id="cb23-8"><a href="qiime-2.html#cb23-8" tabindex="-1"></a>  <span class="at">--o-representative-sequences</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/rep-seqs.qza <span class="dt">\</span></span>
<span id="cb23-9"><a href="qiime-2.html#cb23-9" tabindex="-1"></a>  <span class="at">--o-denoising-stats</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/denoising-stats.qza</span></code></pre></div>
<ul>
<li><code>--i-demultiplexed-seqs</code>: Input file with demultiplexed sequences.</li>
<li><code>--p-trim-left-f</code> and <code>--p-trim-left-r</code>: Number of bases to trim from the start of each read.</li>
<li><code>--p-trunc-len-f</code> and <code>--p-trunc-len-r</code>: Position at which reads are truncated due to decrease in quality.</li>
<li><code>--o-table</code>: Output file for the feature table (ASVs).</li>
<li><code>--o-representative-sequences</code>: Output file for the representative sequences.</li>
<li><code>--o-denoising-stats</code>: Output file for denoising statistics.</li>
</ul>
</div>
<div id="taxonomic-classification" class="section level3 hasAnchor" number="6.3.5">
<h3><span class="header-section-number">6.3.5</span> Taxonomic Classification<a href="qiime-2.html#taxonomic-classification" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This step classifies the representative sequences into taxonomic groups using a pre-trained classifier.</p>
<p><a href="https://docs.qiime2.org/2024.5/plugins/available/feature-classifier/classify-sklearn/">Qiime 2 Docs</a></p>
<div class="sourceCode" id="cb24"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb24-1"><a href="qiime-2.html#cb24-1" tabindex="-1"></a><span class="ex">qiime</span> feature-classifier classify-sklearn <span class="dt">\</span></span>
<span id="cb24-2"><a href="qiime-2.html#cb24-2" tabindex="-1"></a>  <span class="at">--i-classifier</span> /mnt/vstor/SOM_PQHS_LXZ716/Reference/QIIME2_Pre-trained_Classifier/silva-138-99-nb-classifier.qza <span class="dt">\</span></span>
<span id="cb24-3"><a href="qiime-2.html#cb24-3" tabindex="-1"></a>  <span class="at">--i-reads</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/rep-seqs.qza <span class="dt">\</span></span>
<span id="cb24-4"><a href="qiime-2.html#cb24-4" tabindex="-1"></a>  <span class="at">--o-classification</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/taxonomy.qza</span></code></pre></div>
<ul>
<li><code>--i-classifier</code>: Pre-trained classifier file.</li>
<li><code>--i-reads</code>: Input file with representative sequences.</li>
<li><code>--o-classification</code>: Output file with taxonomic classification results.</li>
</ul>
</div>
<div id="metadata-visualization" class="section level3 hasAnchor" number="6.3.6">
<h3><span class="header-section-number">6.3.6</span> Metadata Visualization<a href="qiime-2.html#metadata-visualization" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This step creates a visualization of the taxonomic classification results, allowing you to review the taxonomy assignments.</p>
<p><a href="https://docs.qiime2.org/2024.5/tutorials/metadata/">Qiime 2 Docs</a></p>
<div class="sourceCode" id="cb25"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb25-1"><a href="qiime-2.html#cb25-1" tabindex="-1"></a><span class="ex">qiime</span> metadata tabulate <span class="dt">\</span></span>
<span id="cb25-2"><a href="qiime-2.html#cb25-2" tabindex="-1"></a>  <span class="at">--m-input-file</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/taxonomy.qza <span class="dt">\</span></span>
<span id="cb25-3"><a href="qiime-2.html#cb25-3" tabindex="-1"></a>  <span class="at">--o-visualization</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/taxonomy.qzv</span></code></pre></div>
<ul>
<li><code>--m-input-file</code>: Input file with metadata to be visualized.</li>
<li><code>--o-visualization</code>: Output file for the visualization.</li>
</ul>
<div class="float">
<img src="taxonomy.png" alt="taxonomy.qzv" />
<div class="figcaption">taxonomy.qzv</div>
</div>
</div>
<div id="feature-table-summary" class="section level3 hasAnchor" number="6.3.7">
<h3><span class="header-section-number">6.3.7</span> Feature Table Summary<a href="qiime-2.html#feature-table-summary" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This step summarizes the feature table, providing an overview of feature counts per sample and other summary statistics.</p>
<p><a href="https://docs.qiime2.org/2024.5/tutorials/read-joining/#view-summary-of-deblur-feature-table">Qiime 2 Dosc</a></p>
<div class="sourceCode" id="cb26"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb26-1"><a href="qiime-2.html#cb26-1" tabindex="-1"></a><span class="ex">qiime</span> feature-table summarize <span class="dt">\</span></span>
<span id="cb26-2"><a href="qiime-2.html#cb26-2" tabindex="-1"></a>  <span class="at">--i-table</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/table.qza <span class="dt">\</span></span>
<span id="cb26-3"><a href="qiime-2.html#cb26-3" tabindex="-1"></a>  <span class="at">--o-visualization</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/table.qzv <span class="dt">\</span></span>
<span id="cb26-4"><a href="qiime-2.html#cb26-4" tabindex="-1"></a>  <span class="at">--m-sample-metadata-file</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/processed_metadata.tsv</span></code></pre></div>
<ul>
<li><code>--i-table</code>: Input feature table file.</li>
<li><code>--o-visualization</code>: Output file for the summary visualization.</li>
<li><code>--m-sample-metadata-file</code>: Metadata file associated with the samples.</li>
</ul>
<div class="float">
<img src="table.png" alt="table.qzv - Overview" />
<div class="figcaption">table.qzv - Overview</div>
</div>
<div class="float">
<img src="table_his.png" alt="table.qzv - Interactive Sample Detail" />
<div class="figcaption">table.qzv - Interactive Sample Detail</div>
</div>
</div>
<div id="phylogenetic-tree-construction" class="section level3 hasAnchor" number="6.3.8">
<h3><span class="header-section-number">6.3.8</span> Phylogenetic Tree Construction<a href="qiime-2.html#phylogenetic-tree-construction" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This step aligns the sequences, masks the alignment to remove highly variable positions, and constructs a phylogenetic tree for diversity analyses.</p>
<p><a href="https://docs.qiime2.org/2024.5/plugins/available/phylogeny/align-to-tree-mafft-fasttree/">Qiime 2 Docs</a></p>
<div class="sourceCode" id="cb27"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb27-1"><a href="qiime-2.html#cb27-1" tabindex="-1"></a><span class="ex">qiime</span> phylogeny align-to-tree-mafft-fasttree <span class="dt">\</span></span>
<span id="cb27-2"><a href="qiime-2.html#cb27-2" tabindex="-1"></a>  <span class="at">--i-sequences</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/rep-seqs.qza <span class="dt">\</span></span>
<span id="cb27-3"><a href="qiime-2.html#cb27-3" tabindex="-1"></a>  <span class="at">--o-alignment</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/aligned-rep-seqs.qza <span class="dt">\</span></span>
<span id="cb27-4"><a href="qiime-2.html#cb27-4" tabindex="-1"></a>  <span class="at">--o-masked-alignment</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/masked-aligned-rep-seqs.qza <span class="dt">\</span></span>
<span id="cb27-5"><a href="qiime-2.html#cb27-5" tabindex="-1"></a>  <span class="at">--o-tree</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/unrooted-tree.qza <span class="dt">\</span></span>
<span id="cb27-6"><a href="qiime-2.html#cb27-6" tabindex="-1"></a>  <span class="at">--o-rooted-tree</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/rooted-tree.qza</span></code></pre></div>
<ul>
<li><code>--i-sequences</code>: Input file with representative sequences.</li>
<li><code>--o-alignment</code>: Output file for aligned sequences.</li>
<li><code>--o-masked-alignment</code>: Output file for masked alignment.</li>
<li><code>--o-tree</code>: Output file for unrooted phylogenetic tree.</li>
<li><code>--o-rooted-tree</code>: Output file for rooted phylogenetic tree.</li>
</ul>
</div>
<div id="taxa-bar-plot-visualization" class="section level3 hasAnchor" number="6.3.9">
<h3><span class="header-section-number">6.3.9</span> Taxa Bar Plot Visualization<a href="qiime-2.html#taxa-bar-plot-visualization" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This step creates a bar plot to visualize the taxonomic composition of the samples, providing insights into the relative abundance of different taxa.</p>
<p><a href="https://docs.qiime2.org/2024.5/plugins/available/taxa/barplot/">Qiime 2 Docs</a></p>
<div class="sourceCode" id="cb28"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb28-1"><a href="qiime-2.html#cb28-1" tabindex="-1"></a><span class="ex">qiime</span> taxa barplot <span class="dt">\</span></span>
<span id="cb28-2"><a href="qiime-2.html#cb28-2" tabindex="-1"></a>  <span class="at">--i-table</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/table.qza <span class="dt">\</span></span>
<span id="cb28-3"><a href="qiime-2.html#cb28-3" tabindex="-1"></a>  <span class="at">--i-taxonomy</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/taxonomy.qza <span class="dt">\</span></span>
<span id="cb28-4"><a href="qiime-2.html#cb28-4" tabindex="-1"></a>  <span class="at">--m-metadata-file</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/processed_metadata.tsv <span class="dt">\</span></span>
<span id="cb28-5"><a href="qiime-2.html#cb28-5" tabindex="-1"></a>  <span class="at">--o-visualization</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/taxa-bar-plots.qzv</span></code></pre></div>
<ul>
<li><code>--i-table</code>: Input feature table file.</li>
<li><code>--i-taxonomy</code>: Input file with taxonomic classification.</li>
<li><code>--m-metadata-file</code>: Metadata file associated with the samples.</li>
<li><code>--o-visualization</code>: Output file for the bar plot visualization.</li>
</ul>
<div class="float">
<img src="taxa-bar-plots.png" alt="taxa-bar-plots.qzv" />
<div class="figcaption">taxa-bar-plots.qzv</div>
</div>
</div>
<div id="alpha-rarefaction-analysis" class="section level3 hasAnchor" number="6.3.10">
<h3><span class="header-section-number">6.3.10</span> Alpha Rarefaction Analysis<a href="qiime-2.html#alpha-rarefaction-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Alpha rarefaction analysis helps to understand the diversity within a sample. This step generates a rarefaction curve, which is useful for assessing the sampling depth required to capture the diversity within each sample.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb29-1"><a href="qiime-2.html#cb29-1" tabindex="-1"></a><span class="ex">qiime</span> diversity alpha-rarefaction <span class="dt">\</span></span>
<span id="cb29-2"><a href="qiime-2.html#cb29-2" tabindex="-1"></a>  <span class="at">--i-table</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/table.qza <span class="dt">\</span></span>
<span id="cb29-3"><a href="qiime-2.html#cb29-3" tabindex="-1"></a>  <span class="at">--i-phylogeny</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/rooted-tree.qza <span class="dt">\</span></span>
<span id="cb29-4"><a href="qiime-2.html#cb29-4" tabindex="-1"></a>  <span class="at">--p-max-depth</span> 30000 <span class="dt">\</span></span>
<span id="cb29-5"><a href="qiime-2.html#cb29-5" tabindex="-1"></a>  <span class="at">--m-metadata-file</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/processed_metadata.tsv <span class="dt">\</span></span>
<span id="cb29-6"><a href="qiime-2.html#cb29-6" tabindex="-1"></a>  <span class="at">--o-visualization</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/alpha-rarefaction.qzv</span></code></pre></div>
<ul>
<li><code>--i-table</code>: Input feature table file.</li>
<li><code>--i-phylogeny</code>: Input phylogenetic tree file.</li>
<li><code>--p-max-depth</code>: Maximum rarefaction depth.</li>
<li><code>--m-metadata-file</code>: Metadata file associated with the samples.</li>
<li><code>--o-visualization</code>: Output file for the rarefaction visualization.</li>
</ul>
<p><strong>Choosing <code>--p-max-depth</code>:</strong></p>
<p>The <code>--p-max-depth</code> parameter specifies the maximum sequencing depth for rarefaction. To choose an appropriate value, follow these steps:</p>
<ol style="list-style-type: decimal">
<li>Visualize Sequence Depth Distribution:</li>
</ol>
<ul>
<li>Before setting <code>--p-max-depth</code>, examine the sequencing depth distribution using the <code>table.qzv</code> file. This file can be generated with the following command:</li>
</ul>
<div class="sourceCode" id="cb30"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb30-1"><a href="qiime-2.html#cb30-1" tabindex="-1"></a><span class="ex">qiime</span> feature-table summarize <span class="dt">\</span></span>
<span id="cb30-2"><a href="qiime-2.html#cb30-2" tabindex="-1"></a>  <span class="at">--i-table</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/table.qza <span class="dt">\</span></span>
<span id="cb30-3"><a href="qiime-2.html#cb30-3" tabindex="-1"></a>  <span class="at">--o-visualization</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/table.qzv <span class="dt">\</span></span>
<span id="cb30-4"><a href="qiime-2.html#cb30-4" tabindex="-1"></a>  <span class="at">--m-sample-metadata-file</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/processed_metadata.tsv</span></code></pre></div>
<ul>
<li>Open the resulting <code>table.qzv</code> file in <a href="https://view.qiime2.org/">QIIME 2 View</a> to explore the sequencing depth distribution.</li>
</ul>
<ol start="2" style="list-style-type: decimal">
<li>Determine an Appropriate Depth:</li>
</ol>
<ul>
<li>Look at the <code>"Interactive Sample Detail"</code> and <code>"Frequency per sample"</code> plot in the <code>table.qzv</code> visualization. Select a -<code>-p-max-depth</code> value that includes the majority of your samples. This helps ensure that a sufficient number of samples are included in the analysis.</li>
<li>Avoid setting the depth too high, as this may exclude many samples with lower sequencing depth. Conversely, setting it too low may not fully capture the diversity within each sample.</li>
</ul>
<ol start="3" style="list-style-type: decimal">
<li>Example:</li>
</ol>
<ul>
<li>If the frequency plot shows that most samples have around 20,000 sequences, you might choose a –p-max-depth of 20,000 or slightly lower (e.g., 15,000) to include as many samples as possible while still capturing the diversity.</li>
</ul>
<p>The resulting visualization (<code>alpha-rarefaction.qzv</code>) allows you to explore the alpha diversity metrics at different sampling depths and helps to determine the adequate sequencing depth for your samples.</p>
<div class="float">
<img src="alpha-rarefaction.png" alt="alpha-rarefaction.qzv" />
<div class="figcaption">alpha-rarefaction.qzv</div>
</div>
</div>
<div id="core-metrics-phylogenetic-analysis" class="section level3 hasAnchor" number="6.3.11">
<h3><span class="header-section-number">6.3.11</span> Core Metrics Phylogenetic Analysis<a href="qiime-2.html#core-metrics-phylogenetic-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Core metrics phylogenetic analysis computes a range of diversity metrics, including both alpha and beta diversity metrics. The <code>--p-sampling-depth</code> parameter should be chosen based on the results of the alpha rarefaction analysis.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb31-1"><a href="qiime-2.html#cb31-1" tabindex="-1"></a><span class="ex">qiime</span> diversity core-metrics-phylogenetic <span class="dt">\</span></span>
<span id="cb31-2"><a href="qiime-2.html#cb31-2" tabindex="-1"></a>  <span class="at">--i-phylogeny</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/rooted-tree.qza <span class="dt">\</span></span>
<span id="cb31-3"><a href="qiime-2.html#cb31-3" tabindex="-1"></a>  <span class="at">--i-table</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/table.qza <span class="dt">\</span></span>
<span id="cb31-4"><a href="qiime-2.html#cb31-4" tabindex="-1"></a>  <span class="at">--p-sampling-depth</span> 4000 <span class="dt">\</span></span>
<span id="cb31-5"><a href="qiime-2.html#cb31-5" tabindex="-1"></a>  <span class="at">--m-metadata-file</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/processed_metadata.tsv <span class="dt">\</span></span>
<span id="cb31-6"><a href="qiime-2.html#cb31-6" tabindex="-1"></a>  <span class="at">--o-rarefied-table</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/rarefied-table.qza <span class="dt">\</span></span>
<span id="cb31-7"><a href="qiime-2.html#cb31-7" tabindex="-1"></a>  <span class="at">--o-observed-features-vector</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/observed-features.qza <span class="dt">\</span></span>
<span id="cb31-8"><a href="qiime-2.html#cb31-8" tabindex="-1"></a>  <span class="at">--o-shannon-vector</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/shannon.qza <span class="dt">\</span></span>
<span id="cb31-9"><a href="qiime-2.html#cb31-9" tabindex="-1"></a>  <span class="at">--o-evenness-vector</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/evenness.qza <span class="dt">\</span></span>
<span id="cb31-10"><a href="qiime-2.html#cb31-10" tabindex="-1"></a>  <span class="at">--o-faith-pd-vector</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/faith-pd.qza <span class="dt">\</span></span>
<span id="cb31-11"><a href="qiime-2.html#cb31-11" tabindex="-1"></a>  <span class="at">--o-unweighted-unifrac-distance-matrix</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/unweighted-unifrac-distance-matrix.qza <span class="dt">\</span></span>
<span id="cb31-12"><a href="qiime-2.html#cb31-12" tabindex="-1"></a>  <span class="at">--o-weighted-unifrac-distance-matrix</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/weighted-unifrac-distance-matrix.qza <span class="dt">\</span></span>
<span id="cb31-13"><a href="qiime-2.html#cb31-13" tabindex="-1"></a>  <span class="at">--o-jaccard-distance-matrix</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/jaccard-distance-matrix.qza <span class="dt">\</span></span>
<span id="cb31-14"><a href="qiime-2.html#cb31-14" tabindex="-1"></a>  <span class="at">--o-bray-curtis-distance-matrix</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/bray-curtis-distance-matrix.qza <span class="dt">\</span></span>
<span id="cb31-15"><a href="qiime-2.html#cb31-15" tabindex="-1"></a>  <span class="at">--o-unweighted-unifrac-pcoa-results</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/unweighted-unifrac-pcoa-results.qza <span class="dt">\</span></span>
<span id="cb31-16"><a href="qiime-2.html#cb31-16" tabindex="-1"></a>  <span class="at">--o-weighted-unifrac-pcoa-results</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/weighted-unifrac-pcoa-results.qza <span class="dt">\</span></span>
<span id="cb31-17"><a href="qiime-2.html#cb31-17" tabindex="-1"></a>  <span class="at">--o-jaccard-pcoa-results</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/jaccard-pcoa-results.qza <span class="dt">\</span></span>
<span id="cb31-18"><a href="qiime-2.html#cb31-18" tabindex="-1"></a>  <span class="at">--o-bray-curtis-pcoa-results</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/bray-curtis-pcoa-results.qza <span class="dt">\</span></span>
<span id="cb31-19"><a href="qiime-2.html#cb31-19" tabindex="-1"></a>  <span class="at">--o-unweighted-unifrac-emperor</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/unweighted-unifrac-emperor.qzv <span class="dt">\</span></span>
<span id="cb31-20"><a href="qiime-2.html#cb31-20" tabindex="-1"></a>  <span class="at">--o-weighted-unifrac-emperor</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/weighted-unifrac-emperor.qzv <span class="dt">\</span></span>
<span id="cb31-21"><a href="qiime-2.html#cb31-21" tabindex="-1"></a>  <span class="at">--o-jaccard-emperor</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/jaccard-emperor.qzv <span class="dt">\</span></span>
<span id="cb31-22"><a href="qiime-2.html#cb31-22" tabindex="-1"></a>  <span class="at">--o-bray-curtis-emperor</span> /mnt/vstor/SOM_PQHS_LXZ716/MicroB2/Analysis_QIIME2/bray-curtis-emperor.qzv</span></code></pre></div>
<ul>
<li><code>--i-phylogeny</code>: Input phylogenetic tree file.</li>
<li><code>--i-table</code>: Input feature table file.</li>
<li><code>--p-sampling-depth</code>: Depth at which to subsample the feature table for diversity metrics.</li>
<li><code>--m-metadata-file</code>: Metadata file associated with the samples.</li>
<li><code>--o-rarefied-table</code>: Output rarefied feature table, used for subsequent diversity analyses.</li>
<li><code>--o-observed-features-vector</code>: Output vector of observed features, representing the number of unique features (e.g., species) observed in each sample.</li>
<li><code>--o-shannon-vector</code>: Output vector of Shannon diversity index, a measure of the richness and evenness of species in a community.</li>
<li><code>--o-evenness-vector</code>: Output vector of Pielou’s evenness, indicating how evenly the individuals are distributed across the different species present.</li>
<li><code>--o-faith-pd-vector</code>: Output vector of Faith’s Phylogenetic Diversity, a measure of diversity that considers the phylogenetic relationships between species.</li>
<li><code>--o-unweighted-unifrac-distance-matrix</code>: Output unweighted UniFrac distance matrix, a measure of beta diversity that considers the presence/absence of species and their phylogenetic relationships.</li>
<li><code>--o-weighted-unifrac-distance-matrix</code>: Output weighted UniFrac distance matrix, similar to unweighted UniFrac but also considering the relative abundance of species.</li>
<li><code>--o-jaccard-distance-matrix</code>: Output Jaccard distance matrix, a measure of beta diversity based on the presence/absence of species.</li>
<li><code>--o-bray-curtis-distance-matrix</code>: Output Bray-Curtis distance matrix, a measure of beta diversity that considers the relative abundance of species.</li>
<li><code>--o-unweighted-unifrac-pcoa-results</code>: Output PCoA results for unweighted UniFrac, used for visualizing the beta diversity.</li>
<li><code>--o-weighted-unifrac-pcoa-results</code>: Output PCoA results for weighted UniFrac, used for visualizing the beta diversity.</li>
<li><code>--o-jaccard-pcoa-results</code>: Output PCoA results for Jaccard distance, used for visualizing the beta diversity.</li>
<li><code>--o-bray-curtis-pcoa-results</code>: Output PCoA results for Bray-Curtis distance, used for visualizing the beta diversity.</li>
<li><code>--o-unweighted-unifrac-emperor</code>: Output Emperor visualization for unweighted UniFrac, an interactive visualization tool.</li>
<li><code>--o-weighted-unifrac-emperor</code>: Output Emperor visualization for weighted UniFrac, an interactive visualization tool.</li>
<li><code>--o-jaccard-emperor</code>: Output Emperor visualization for Jaccard distance, an interactive visualization tool.</li>
<li><code>--o-bray-curtis-emperor</code>: Output Emperor visualization for Bray-Curtis distance, an interactive visualization tool.</li>
</ul>
<p><strong>Choosing –p-sampling-depth:</strong></p>
<p>The <code>--p-sampling-depth</code> parameter should be chosen based on the results of the alpha rarefaction analysis. In the alpha rarefaction visualization <code>(alpha-rarefaction.qzv)</code>, look at the sequencing depth at which the observed features plateau for most samples. This depth ensures that the diversity metrics are calculated with an adequate number of sequences without excluding too many samples. For example, if the observed features plateau around 4,000 sequences, you might set <code>--p-sampling-depth</code> to 4,000.</p>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="samtools.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="kraken2.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/YiqianZhang1004/hpc_tutorialindex.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
